mayastor/src/bdev/nexus/nexus_metadata.rs

```diff
@@ -0,0 +1,620 @@
+pub enum MetaDataError {
+    #[snafu(display("Incorrect MetaData header size"))]
+    HeaderSize {},
+    #[snafu(display("Incorrect MetaData header signature"))]
+    HeaderSignature {},
+    #[snafu(display("Incorrect MetaData header checksum"))]
+    HeaderChecksum {},
+    #[snafu(display("Incorrect MetaData index checksum"))]
+    IndexChecksum {},
+    #[snafu(display("Incorrect MetaData configuration object checksum"))]
+    ObjectChecksum {},
+
+    #[snafu(display("MetaData serialization error"))]
+    SerializeError { source: Error },
+    #[snafu(display("MetaData deserialization error"))]
+    DeserializeError { source: Error },
+
+    #[snafu(display("Failed to allocate index: {}", source))]
+    IndexAlloc { source: DmaError },
+    #[snafu(display("Failed to write index: {}", source))]
+    IndexWrite { source: ChildIoError },
+
+    #[snafu(display("MetaData index is inconsistent"))]
+    IndexInconsistent {},
+    #[snafu(display(
+        "Number of objects {} exceeds index size {}",
+        count,
+        size
+    ))]
+    IndexSizeExceeded { count: u32, size: u32 },
+    #[snafu(display("Not enough space left on MetaData partition"))]
+    PartitionSizeExceeded {},
+
+    #[snafu(display("Failed to allocate configuration object: {}", source))]
+    ObjectAlloc { source: DmaError },
+    #[snafu(display("Failed to read configuration object: {}", source))]
+    ObjectRead { source: ChildIoError },
+    #[snafu(display("Failed to write configuration object: {}", source))]
+    ObjectWrite { source: ChildIoError },
+
+    #[snafu(display("MetaData partition is missing or invalid"))]
+    MissingPartition {},
+
+    #[snafu(display("Error calculating timestamp: {}", source))]
+    TimeStampError { source: SystemTimeError },
+
+    #[snafu(display("{}", error))]
+    NexusChildError { error: String },
+}
+
+#[derive(Debug, Deserialize, PartialEq, Default, Serialize, Copy, Clone)]
+pub struct MetaDataHeader {
+    /// Signature identifying this as a MetaDataHeader object
+    pub signature: [u8; 8],
+    /// Size of this header in bytes
+    pub header_size: u32,
+    /// CRC-32 checksum of this header
+    pub self_checksum: u32,
+    /// Current object generation counter
+    pub generation: u64,
+    /// Absolute location (LBA) of this header on disk
+    pub self_lba: u64,
+    /// Offset of start of index table relative to self_lba
+    pub index_start: u64,
+    /// Number of valid entries in the index
+    pub used_entries: u32,
+    /// Maximum number of entries that the index can contain
+    pub max_entries: u32,
+    /// Size of an index entry in bytes
+    pub entry_size: u32,
+    /// CRC-32 checksum of the index table
+    pub index_checksum: u32,
+    /// Offset of first usable data block relative to self_lba
+    pub data_start: u64,
+    /// Offset of last usable data block relative to self_lba
+    pub data_end: u64,
+}
+
+impl MetaDataHeader {
+    pub const METADATA_HEADER_SIZE: u32 = 72;
```

Comment:
This maps to 136 bytes? .. orso? what about the others i.e 512-136? are they reserved? should we pad the struct to 512 and mark it reserved? I think thats wise to do...